frontend/pkg/frontend/frontend.go

```diff
@@ -86,11 +86,24 @@ func (f *Frontend) Run(ctx context.Context, stop <-chan struct{}) {
 	// This just digs up the logger passed to NewFrontend.
 	logger := LoggerFromContext(f.server.BaseContext(f.listener))
 
+	errs, ctx := errgroup.WithContext(ctx)
+	errs.Go(func() error {
+		shutdown, err := InstallOpenTelemetryTracer(ctx, logger)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := shutdown(ctx); err != nil {
+				logger.Error("otel-sdk shutdown failed", "error", err)
+			}
+		}()
+		return nil
+	})
```

Comment:
Forgive me if this is a dumb question but when does `shutdown` actually run here?

Since this is now in a goroutine and `InstallOpenTelemetryTracer` doesn't block, I'm concerned that `shutdown` will just get called immediately since it looks like the goroutine exits immediately.

Either I'm misunderstanding the semantics of `errgroup` or maybe `InstallOpenTelemetryTracer` should just be called directly in the `Run` method.

Thoughts?

Reply:
y, it got called immediately and skilled any trace export. I moved the shutdown call to the stop routine.